tasks/main.yml

```diff
@@ -242,19 +242,19 @@
     "SQL Server needs to be restarted in order to apply this setting."
     in __mssql_conf_set_tcp_port.stdout
   notify: Restart the mssql-server service
-  when: mssql_tcp_port is defined
+  when: mssql_tcp_port is not none
 
 - name: Verify that the mssql_password variable is defined
   assert:
     that:
-      - mssql_password is defined
+      - mssql_password is not none
     fail_msg: >-
       You must define the mssql_password variable because MSSQL requires
       the sa user to authenticate to input SQL files.
-  when: mssql_input_sql_file is defined
+  when: mssql_input_sql_file is not none
```

Comment:
I missed that initially - it is redundant to do when: on assert: that has the same condition. assert: checks the condition itself.